### ISSUE TITLE
Air SDK 33.1.1.345, got error:90034 while uploading ipa to app store

### ISSUE THREAD

**Issue by wujunas** [Gamua/Adobe-Runtime-Support#676]
Hi,
I created an ipa file to upload it to app store, with the new Air SDK 33.1.1.345. got an error message: "ERROR ITMS-90034: "Missing or invalid signature. The bundle 'com.yuan.xxx' at bundle path 'Payload/xxx.app' is not signed using an Apple submission certificate.".

The certificate I use is newly created and has not expired.Anyone had this issue? knows what to do?

**Comment by coma1229** [comment]
Hi,
I also encountered the same problem.
However, I re-signed the .ipa file on my Mac's Terminal app and was able to resolve the issue.
I hope it helps you solve your problem.
(I'm sorry I'm not good at English. )

**Comment by georgemck** [comment]
Yes, I did the same on my Mac, which has XCode installed. I made a shell script called packager.sh with the below information and used chmod +x packager.sh to make it executable. Calling it with ./packager.sh. This saved me time later on.

Instructions from here: 

# https://stackoverflow.com/questions/64596362/iphone-distribution-certificate-is-not-trusted

# this version was tested OK vith macOs Sierra 10.12.5 (16F73) on oct 0th, 2017
# original ipa file must be store in current working directory

IPA="ipa-filename.ipa"
PROVISION="path-to.mobileprovision"
CERTIFICATE="hexadecimal-certificate-identifier" # must be in keychain
# identifier maybe retrieved by running: security find-identity -v -p codesigning

# unzip the ipa
unzip -q "$IPA"

# remove the signature
rm -rf Payload/*.app/_CodeSignature

# replace the provision
cp "$PROVISION" Payload/*.app/embedded.mobileprovision

# generate entitlements for current app
cd Payload/
codesign -d --entitlements - *.app > entitlements.plist
cd ..
mv Payload/entitlements.plist entitlements.plist

# sign with the new certificate and entitlements
/usr/bin/codesign -f -s "$CERTIFICATE" '--entitlements' 'entitlements.plist' Payload/*.app

# zip it back up
zip -qr resigned.ipa Payload

**Comment by kasparv** [comment]
Started happening for me to after I started using new iOS distribution certificate. 
Fortunately, I was able to do the signing manually by followint the steps above.
In case anyone wonders how to find the hexadecimal certificate identifier, use the following command:
`security find-identity -p codesigning`

**Comment by ajwfrost** [comment]
I believe this is the same root cause as the other code-signing/IPA installation issues .. a patch `adt.jar` file is available from the below link and we'll get an updated SDK out asap..
https://transfer.harman.com/message/4SPAINrOXZzGEDmaAEUqNF
thanks

**Comment by leossmith** [comment]
> Yes, I did the same on my Mac, which has XCode installed. I made a shell script called packager.sh with the below information and used chmod +x packager.sh to make it executable. Calling it with ./packager.sh. This saved me time later on.
> 
> Instructions from here:
> 
> # https://stackoverflow.com/questions/64596362/iphone-distribution-certificate-is-not-trusted
> # this version was tested OK vith macOs Sierra 10.12.5 (16F73) on oct 0th, 2017
> # original ipa file must be store in current working directory
> IPA="ipa-filename.ipa"
> PROVISION="path-to.mobileprovision"
> CERTIFICATE="hexadecimal-certificate-identifier" # must be in keychain
> 
> # identifier maybe retrieved by running: security find-identity -v -p codesigning
> # unzip the ipa
> unzip -q "$IPA"
> 
> # remove the signature
> rm -rf Payload/*.app/_CodeSignature
> 
> # replace the provision
> cp "$PROVISION" Payload/*.app/embedded.mobileprovision
> 
> # generate entitlements for current app
> cd Payload/
> codesign -d --entitlements - *.app > entitlements.plist
> cd ..
> mv Payload/entitlements.plist entitlements.plist
> 
> # sign with the new certificate and entitlements
> /usr/bin/codesign -f -s "$CERTIFICATE" '--entitlements' 'entitlements.plist' Payload/*.app
> 
> # zip it back up
> zip -qr resigned.ipa Payload

is there a way to do that on windows?

**Comment by ajwfrost** [comment]
Closing this issue as we believe we have now got to a point with the latest AIR SDK where all the code-signing is working for iOS on both mac and windows platforms... for these sorts of problems, it may be best to raise a new, specific issue if there is still a problem with it.
thanks
